popsycle/synthetic.py

```diff
@@ -3512,3 +3527,390 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+    rho_knot = (200/3)*(rho_crit)*((c**3)/(np.log(c+1)-(c/(1+c)))) # Msun/Mpc^3
+    mass_8_kpc = (4*np.pi*rho_knot*(r_s**3)*(np.log((r_s + r_max)/r_s) - (r_max/(r_s+r_max))))*(((10**3)**3)/((10**6)**3))*fdm #Msun
+
+    #Determine the number of PBHs within that distance
+    num_pbh_8kpc = (mass_8_kpc/pbh_mass)
+    num_pbh_8kpc = round(num_pbh_8kpc, -1)
+
+    #Defining needed functions from the python package "NFWdist".
+    #Used to calculate PBH radii from galactic center assuming NFW profile.
+    #https://github.com/CullanHowlett/NFWdist
+    def pnfwunorm(q, con=5):
+        if hasattr(con, '__len__'):
+            y = np.outer(q,con)
+        else:
+            y = q*con
+        return np.log(1.0 + y)-y/(1.0 + y)
+
+    def pnfw(q, con=5, logp=False):
+        p = pnfwunorm(q, con=con)/pnfwunorm(1, con=con)
+        if hasattr(q, '__len__'):
+            p[q>1] = 1
+            p[q<=0] = 0
+        else:
+            if (q > 1):
+                p = 1
+            elif (q <= 0):
+                p = 0
+        if(logp):
+            return np.log(p)
+        else:
+            return p
+
+    def qnfw(p, con=5, logp=False):
+        if (logp):
+            p = np.exp(p)
+        if hasattr(p, '__len__'):
+            p[p>1] = 1
+            p[p<=0] = 0
+        else:
+            if (p > 1):
+                p = 1
+            elif (p <= 0):
+                p = 0
+        if hasattr(con, '__len__'):
+            p = np.outer(p,pnfwunorm(1, con=con))
+        else:
+            p *= pnfwunorm(1, con=con)
+        return (-(1.0/np.real(special.lambertw(-np.exp(-p-1))))-1)/con
+
+    #Calculating radius values for all PBHs within 8.3kpc of the galactic center
+    r_values = (qnfw(np.random.rand(int(num_pbh_8kpc)) * pnfw(r_max/r_vir,con=c, logp=False), con=c)*r_vir)
+
+    #Sample PBH lattitude and longitude to have full spherical coordinates.
+    sin_lats = np.random.uniform(-1, 1, int(num_pbh_8kpc))
+    lats=np.arcsin(sin_lats)
+    longs = np.random.uniform(0, 2*np.pi, int(num_pbh_8kpc))
+
+    #Converting spherical galactocentric coordinates to cartesian galactocentric coordinates
+    cart = astropy.coordinates.spherical_to_cartesian(r_values, lats, longs)
+
+    #Defining galactocentric coordinates
+    c = coord.Galactocentric(x=cart[0] * units.kpc, y=cart[1] * units.kpc, z=cart[2] * units.kpc)
+
+    #Transforming from galactocentric to galactic coordinates
+    #outputs l, b, dist in degrees.
+    c = c.transform_to(coord.Galactic(representation_type='cartesian'))
+
+    #Set minimum and maximum l, b, and r values for PBH mask. 
+    #Determining the center of the field of view circle.
+    l_min = np.min(long_bin).values
+    l_max = np.max(long_bin).values
+    b_min = np.min(lat_bin).values
+    b_max = np.max(lat_bin).values
+    l_mid = np.average((l_min, l_max))
+    b_mid = np.average((b_min, b_max))
+
+    dists = angdist(l_mid, b_mid, c.l.deg, c.b.deg)
+    r_max_mask = 16.6
```

Comment:
What is this here for?

Reply:
I will change it to 2*r_max. Thank you. It is twice r_max because at the point of the mask the radii are in galactic/heliocentric coordinates. Will said that we need to consider objects further than the galactic center as there could be sources or lenses back there.